base/server/src/main/java/org/dogtagpki/server/cli/SubsystemIdGeneratorUpdateCLI.java

```diff
@@ -0,0 +1,310 @@
+//
+// Copyright Red Hat, Inc.
+//
+// SPDX-License-Identifier: GPL-2.0-or-later
+//
+package org.dogtagpki.server.cli;
+
+import com.netscape.certsrv.base.EBaseException;
+import com.netscape.cmscore.apps.CMS;
+import com.netscape.cmscore.apps.DatabaseConfig;
+import com.netscape.cmscore.apps.EngineConfig;
+import com.netscape.cmscore.dbs.DBSubsystem;
+import com.netscape.cmscore.dbs.Repository;
+import com.netscape.cmscore.dbs.Repository.IDGenerator;
+import static com.netscape.cmscore.dbs.Repository.logger;
+import com.netscape.cmscore.ldapconn.LDAPConfig;
+import com.netscape.cmscore.ldapconn.LDAPConnectionConfig;
+import com.netscape.cmscore.ldapconn.LdapAuthInfo;
+import com.netscape.cmscore.ldapconn.LdapBoundConnection;
+import com.netscape.cmscore.ldapconn.LdapConnInfo;
+import com.netscape.cmscore.ldapconn.PKISocketConfig;
+import com.netscape.cmscore.ldapconn.PKISocketFactory;
+import com.netscape.cmscore.request.RequestRepository;
+import com.netscape.cmsutil.password.PasswordStore;
+import com.netscape.cmsutil.password.PasswordStoreConfig;
+import java.math.BigInteger;
+import netscape.ldap.LDAPAttribute;
+import netscape.ldap.LDAPAttributeSet;
+import netscape.ldap.LDAPEntry;
+import netscape.ldap.LDAPException;
+import netscape.ldap.LDAPModification;
+import netscape.ldap.LDAPSearchResults;
+import netscape.ldap.LDAPv3;
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.Option;
+import org.dogtagpki.cli.CLI;
+import org.dogtagpki.util.logging.PKILogger;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * @author Marco Fargetta {@literal <mfargett@redhat.com>}
+ */
+public abstract class SubsystemIdGeneratorUpdateCLI extends SubsystemCLI {
+    private static final Logger logger = LoggerFactory.getLogger(SubsystemIdGeneratorUpdateCLI.class);
+    protected IDGenerator idGenerator;
+
+    public SubsystemIdGeneratorUpdateCLI(CLI parent) {
+        super("update", "Update " + parent.getParent().getParent().getName().toUpperCase() + " range generator", parent);
+    }
+    @Override
+    public void createOptions() {
+        options.addOption("t", "type", true, "Generator type to update.");
+    }
+
+    @Override
+    public void execute(CommandLine cmd) throws Exception {
+
+        if (!cmd.hasOption("type")) {
+            throw new Exception("Missing generator type.");
+        }
+        String generatorType = cmd.getOptionValue("type");
+
+        if (cmd.hasOption("debug")) {
+            PKILogger.setLevel(PKILogger.LogLevel.DEBUG);
+
+        } else if (cmd.hasOption("verbose")) {
+            PKILogger.setLevel(PKILogger.LogLevel.INFO);
+        }
+
+        String[] cmdArgs = cmd.getArgs();
+
+        if (cmdArgs.length != 1) {
+            throw new Exception("Missing generator");
+        }
```

Comment:
Could we flip the option and the positional argument? I think usually the object we want to change should be specified as a positional argument, then the attribute that we want to change should be specified as an option (because there could be other attributes that we want to to change in the future).

So instead of:

```
$ pki-server ca-id-generator-update --type cert legacy2
```

it will be:

```
$ pki-server ca-id-generator-update --type legacy2 cert
```

Reply:
Thanks for the update!